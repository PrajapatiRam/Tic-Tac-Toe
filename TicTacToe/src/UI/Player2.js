import React, {Component} from 'react';
import {connect} from 'react-redux';

import {StyleSheet, View, TextInput, Button, Text} from 'react-native';
import Header from './Header';
import {StackActions, NavigationActions} from 'react-navigation';
import {savePlayer2Name} from '../Redux/Actions/AppActions';

class Player2 extends Component {
  constructor() {
    super();
    this.state = {
      player2Name: 'Player 2',
    };
  }

  render() {
    return (
      <View style={styles.container}>
        <Header title={'Welcome'} />
        <Text style={styles.text}>{'Player 2'}</Text>
        <TextInput
          style={styles.textInput}
          placeholder={'Your name?'}
          onChangeText={player2Name => {
            this.setState({player2Name});
          }}
        />
        <Button
          title={'Start'}
          onPress={() => {
            this.saveName();
            const resetAction = StackActions.reset({
              index: 0,
              actions: [NavigationActions.navigate({routeName: 'HomeScene'})],
            });
            this.props.navigation.dispatch(resetAction);
          }}
        />
      </View>
    );
  }

  saveName() {
    this.props.savePlayer2Name(this.state.player2Name);
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    padding: 20,
    backgroundColor: '#F5FCee',
  },
  textInput: {
    borderColor: 'black',
    borderWidth: 1,
    marginVertical: 25,
  },
  text: {
    marginTop: 25,
    alignSelf: 'center',
    fontSize: 20,
    fontWeight: 'bold',
  },
});

export default connect(
  null,
  {
    savePlayer2Name: savePlayer2Name,
  },
)(Player2);
