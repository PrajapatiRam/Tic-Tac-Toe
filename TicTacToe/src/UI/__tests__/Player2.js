import React from 'react';
import Enzyme, {shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';
import mockStore from 'redux-mock-store';
import Player2 from '../Player2';
Enzyme.configure({adapter: new Adapter()});
it('Player2 renders as expected', () => {
  const initialState = {
    player1name: 'test Player1',
    player2name: 'test Player2',
    player1score: 1,
    player2score: 2,
  };
  const store = mockStore(initialState);

  const wrapper = shallow(<Player2 store={store} />);

  const component = shallow(wrapper.getElements()[0]);

  expect(toJson(component)).toMatchSnapshot();
});
