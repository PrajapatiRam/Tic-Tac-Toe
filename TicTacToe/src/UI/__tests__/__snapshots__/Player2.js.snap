// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Player2 renders as expected 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#F5FCee",
      "flex": 1,
      "padding": 20,
    }
  }
>
  <Header
    title="Welcome"
  />
  <Text
    style={
      Object {
        "alignSelf": "center",
        "fontSize": 20,
        "fontWeight": "bold",
        "marginTop": 25,
      }
    }
  >
    Player 2
  </Text>
  <TextInput
    allowFontScaling={true}
    onChangeText={[Function]}
    placeholder="Your name?"
    rejectResponderTermination={true}
    style={
      Object {
        "borderColor": "black",
        "borderWidth": 1,
        "marginVertical": 25,
      }
    }
    underlineColorAndroid="transparent"
  />
  <Button
    onPress={[Function]}
    title="Start"
  />
</View>
`;
